packages/opentelemetry-tracing/src/BasicTracerFactory.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { BasicTracerConfig } from './types';
+import { BasicTracer } from './BasicTracer';
+import { SpanProcessor } from './SpanProcessor';
+
+export class BasicTracerFactory implements types.TracerFactory {
+  private static _singletonInstance: BasicTracerFactory;
+  private readonly _tracers: Map<String, BasicTracer> = new Map();
+  private _spanProcessors: SpanProcessor[] = [];
+  private _config?: BasicTracerConfig;
+
+  constructor(config?: BasicTracerConfig) {
```

Comment:
constructor could be private, forcing users to use `BasicTracerFactory.instance` which would prevent creation of multiple NodeTracerFactories. Is there a use case where we would want to allow the creation of multiple or should it always be a singleton?